### PR TITLE
Enable support for EKS IRSA

### LINKED ISSUE #2703: Support IAM for Service Accounts in AWS
Output of `restic version`
--------------------------
restic 0.9.6 compiled with go1.13.4 on linux/amd64

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
Restic should support assuming role via WebIdentityTokenFile, which was added to minio (https://github.com/minio/minio-go/pull/1183). Doing so restic would be able to use a service account that is using AWS's eks.amazonaws.com/role-arn annotation to attach a role to it. This would be a big benefit since there wouldn't be a need for user credentials that have to be rolled regularly.

What are you trying to do?
--------------------------
I'd like to create backups of PVs in AWS that are stored in S3 bucket, without having an additional IAM user that I've to maintain.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
The backup with restic it self works great, but having the additional user creates a lot overhead for us.

### PR DESCRIPTION
**Pull request by ac-hibbert**
What is the purpose of this change? What does it change?
--------------------------------------------------------

This change enables restic use via the EKS IAM roles for service accounts feature (https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) this is enabled by updating the minio-go version to v6.0.53 (see https://github.com/minio/minio-go/releases) which contains the following PRs:

https://github.com/minio/minio-go/pull/1263 - in v6.0.53
https://github.com/minio/minio-go/pull/1183 - in v6.0.45

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

This was requested in and closes #2703

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: changelog/unreleased/pull-2733 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Enable support for EKS IRSA
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-2733:1]: Actually this PR updates the minio library used for S3-like back-ends. And as a side-effect we get support for AWS EKS IRSA (hmm, that are too many abbreviations at once).

I would suggest to call this `S3 backend: Add support for WebIdentityTokenFile`.

File: changelog/unreleased/pull-2733 [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+This change enables restic to use S3 as a backend using the EKS IAM Roles For
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-2733:3]: The official capitalization seems to be "IAM roles for service accounts". I'd reword the sentence to "We've added support for EKS IAM roles for service accounts feature to the S3 backend."

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
Please adapt the changelog according to my comments and fix the test failures.

Edit: Did you or someone else test this PR manually?

**Comment by hibbert** [comment]
This was tested manually in an AWS EKS cluster.

I have made the changes you have mentioned, but the build still seems to be failing. I think I'll need further input on why this is.

File: changelog/unreleased/pull-2733 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+S3 backend: Add support for WebIdentityTokenFile
```
Comment (greatroar) [inline changelog/unreleased/pull-2733:1]: The build fails because of this line. It must start with Bugfix:, Change:, Enhancement: or Security:. In this case, I guess it should read:

    Enhancement: Add support for WebIdentityTokenFile in S3 backend

Comment (hibbert) [reply changelog/unreleased/pull-2733]: I have changed this as advised

**Comment by roman-judin-dragon** [comment]
@hibbert seems this is the root cause of failing of [run_integration_tests.go](https://travis-ci.com/github/restic/restic/jobs/342129000)
>../../../../pkg/mod/github.com/minio/minio-go/v6@v6.0.53/api.go:569:16: c.httpClient.CloseIdleConnections undefined (type *http.Client has no field or method CloseIdleConnections)

**Comment by MichaelEischer** [comment]
Minio 6.0.53 seems to depend on at least Go 1.12. The `c.httpClient.CloseIdleConnections` method is only supported since Go 1.12. I'm not sure whether we're ready to drop support for Go 1.12 yet.

**Comment by hibbert** [comment]
Thanks for your reply. I submitted this so that I can use EKS IRSA with a 3rd party component that uses restic which is called velero (issue - https://github.com/vmware-tanzu/velero/issues/2457). I know that a similar application requires this change called stash (issue - https://github.com/stashed/stash/issues/1101). Hopefully this can be resolved soon.

**Comment by greatroar** [comment]
@MichaelEischer Go 1.11 no longer receives security updates.

**Comment by MichaelEischer** [comment]
I've rebased this PR to the current master and squashed all commits.

**Review by MichaelEischer — APPROVED** [review]
LGTM

**Comment by galindro** [comment]
@MichaelEischer is there any ETA for this feature be released?

**Comment by rawtaz** [comment]
@galindro As it has now been merged into master, it will be included in the next restic release, for which there's no formal ETA that we can communicate. It will be soonish.

**Comment by galindro** [comment]
ok tks
